apps/passport-server/src/services/telegramService.ts

```diff
@@ -152,28 +113,34 @@ export class TelegramService {
           logger(
             `[TELEGRAM] Approving chat join request for ${userId} to join ${chatId}`
           );
+          const chat = (await ctx.api.getChat(chatId)) as TopicChat;
+
           await this.bot.api.sendMessage(
             userId,
             `<i>Verifying and inviting...</i>`,
             { parse_mode: "HTML" }
           );
           await this.bot.api.approveChatJoinRequest(chatId, userId);
-          const chat = (await this.bot.api.getChat(
-            chatId
-          )) as Chat.GroupGetChat;
-          const inviteLink = await ctx.createChatInviteLink();
```

Comment:
Instead of creating a new invite link that can be forwarded and used by any account, we just give the user the same link that just got approved. I have tested and forwarding an approved link doesn't allow the recipient to join without also getting approval.